Floofbot/Modules/Logging.cs

```diff
@@ -177,226 +135,297 @@ public async Task Toggle()
+                            break;
+                        case "UserBannedChannel":
+                            logChannel = serverConfig.UserBannedChannel;
+                            break;
+                        case "UserUnbannedChannel":
+                            logChannel = serverConfig.UserUnbannedChannel;
+                            break;
+                        case "UserJoinedChannel":
+                            logChannel =  serverConfig.UserJoinedChannel;
+                            break;
+                        case "UserLeftChannel":
+                            logChannel = serverConfig.UserLeftChannel;
+                            break;
+                        case "MemberUpdatesChannel":
+                            logChannel =  serverConfig.MemberUpdatesChannel;
+                            break;
+                        case "UserKickedChannel":
+                            logChannel =  serverConfig.UserKickedChannel;
+                            break;
+                        case "UserMutedChannel":
+                            logChannel =  serverConfig.UserMutedChannel;
+                            break;
+                        case "UserUnmutedChannel":
+                            logChannel =  serverConfig.UserUnmutedChannel;
+                            break;
+                        default:
+                            logChannel = 0;
+                            break;
                     }
+                    var textChannel = await guild.GetTextChannelAsync(logChannel);
+                    return textChannel;
                 }
+                return null;
             }
             public bool IsToggled(IGuild guild)
             {
                 // check if the logger is toggled on in this server
                 // check the status of logger
-                string sqlCheckStatus = $"SELECT IsOn FROM Logger WHERE ServerID = {guild.Id} LIMIT 1"; // check state
-                using (SqliteCommand command = new SqliteCommand(sqlCheckStatus, dbConnection))
-                {
-                    using (SqliteDataReader result = command.ExecuteReader())
-                    {
-                        while (result.Read())
-                        {
-                            long bEnabled = (long)result["IsOn"];
-                            if (bEnabled == 0)
-                                return false;
-                            else if (bEnabled == 1)
-                                return true;
-                            else
-                                return false;
-                        }
-                        return false;
-                    }
-                }
+                var ServerConfig = _floofDb.LogConfigs.Find(guild.Id);
+                if (ServerConfig == null) // no entry in DB for server - not configured
+                    return false;
+
+                bool bEnabled = ServerConfig.IsOn;

```

Comment:
:foxsockangry:

Reply:
whats the issue with this?

Reply:
This line can be changed to `return ServerConfig.IsOn;`, and the remaining lines can be removed. Again, seems like an artifact of previous design patterns.

Although having never worked with entity frameworks before, you may need to employ a cast to `bool` to keep the type checker happy.

Reply:
oh I see, Agreed. Will do this.